src/MiddlewareListener.php

```diff
@@ -69,7 +80,13 @@ public function onDispatch(MvcEvent $event)
             foreach ($routeMatch->getParams() as $key => $value) {
                 $psr7Request = $psr7Request->withAttribute($key, $value);
             }
-            $return = $middleware($psr7Request, Psr7Response::fromZend($response));
+            $return = $pipe(
```

Comment:
Ah, okay, nice.

As noted, using `process()` will be forwards compatible, meaning that when Stratigility adopts a finalized PSR-15, we likely can just add an `||` to the constraint and not need any actual code changes. I figure, do it once. :smile: